### PR TITLE
replaygain: target level refactor

### PR DESCRIPTION
**Pull request by zsinskri**
Remove REPLAYGAIN_* vs R128_* tag handling from the Backends, restrict it to the main plugin logic. That is archieved by passing the target loudness level to the backend with every album/track. It also necessitates that all Backends properly implement the target level.
As the different tag formats call for different default target levels this also seperates the configuration of those levels.

### CODE REVIEW

**Comment by zsinskri** [comment]
The latest push merges master (or rather #3056, from which this branch was forked), resolving conflicts. It also fixes the `ReplayGainLdnsCliMalformedTest`, which caused unrelated tests to fail, due to missing cleanup.

**Comment by zsinskri** [comment]
Is there any guideline on when to merge and when to rebase for this project? I will rework this branch's structure if a rebase is preferred.

**Comment by zsinskri** [comment]
Rebased onto master.

**Comment by zsinskri** [comment]
#3056 is merged. This is now ready for review.

File: beetsplug/replaygain.py [hunk 1]
```diff
@@ -232,6 +267,12 @@ def compute_chunk_gain(self, items, is_album):
 
         self._log.debug(u'analysis finished: {0}', output)
         results = self.parse_tool_output(output, path_list, is_album)
+
+        if gain_adjustment != 0:
+            for i in range(len(results)):
+                orig = results[i]
+                results[i] = Gain(orig.gain + gain_adjustment, orig.peak)
```
Comment (sampsyo) [inline beetsplug/replaygain.py]: Can be simplified a bit:

```py
for i, orig in enumerate(results):
    results[i] = Gain(orig.gain + gain_adjustment, orig.peak)
```

File: beetsplug/replaygain.py [hunk 2]
```diff
@@ -232,6 +267,12 @@ def compute_chunk_gain(self, items, is_album):
 
         self._log.debug(u'analysis finished: {0}', output)
         results = self.parse_tool_output(output, path_list, is_album)
+
+        if gain_adjustment != 0:
```
Comment (sampsyo) [inline beetsplug/replaygain.py]: Zero integers are falsy in Python:
```suggestion
        if gain_adjustment:
```

File: beetsplug/replaygain.py [hunk 3]
```diff
@@ -436,29 +462,36 @@ def _construct_cmd(self, item, peak_method):
...
 
         If `count_blocks` is False, the number of gating blocks returned
         will be 0.
         """
+        target_level_lufs = db_to_lufs(target_level)
+        peak_method = {
+            Peak.none: "none",
+            Peak.true: "true",
+            Peak.sample: "sample",
+        }[peak]
```
Comment (sampsyo) [inline beetsplug/replaygain.py]: Seems like `peak.name` would also work and avoid the lookup table?

File: beetsplug/replaygain.py [hunk 4]
```diff
@@ -436,29 +462,36 @@ def _construct_cmd(self, item, peak_method):
...
+        cmd = self._construct_cmd(item, peak_method)
         self._log.debug(
             u'executing {0}', u' '.join(map(displayable_path, cmd))
         )
         output = call(cmd).stderr.splitlines()
 
         # parse output
 
-        if self._peak_method == "none":
+        if peak is Peak.none:
```
Comment (sampsyo) [inline beetsplug/replaygain.py]: The [Python docs](https://docs.python.org/3/library/enum.html) suggest that enum values should be compared with equality (`==`), not identity (`is`).

File: docs/changelog.rst [hunk 5]
```diff
@@ -42,6 +42,13 @@ New features:
   new ``discogs_albumid`` field from the Discogs API.
   Thanks to :user:`thedevilisinthedetails`.
   :bug:`465` :bug:`3322`
+* :doc:`plugins/replaygain`: ``r128_targetlevel`` is a new configuration option
+  for the ReplayGain plugin: It defines the reference volume for files using
+  ``R128_`` tags. ``targtelevel`` only configures the reference volume for
```
Comment (sampsyo) [inline docs/changelog.rst]:
```suggestion
  ``R128_`` tags. ``targetlevel`` only configures the reference volume for
```

File: docs/plugins/replaygain.rst [hunk 6]
```diff
@@ -120,6 +125,13 @@ This option only works with the "ffmpeg" backend:
 - **peak**: Either ``true`` (the default) or ``sample``. ``true`` is
   more accurate but slower.
 
+This option is deprecated:
+
+- **method**: The loudness scanning standard: either `replaygain` for
+  ReplayGain 2.0, `ebu` for EBU R128, or `atsc` for ATSC A/85. This dictates
+  the reference level: -18, -23, or -24 LUFS respectively. Only supported by
+  "bs1770gain" backend.
```
Comment (sampsyo) [inline docs/plugins/replaygain.rst]: I think it's fine to just leave this one off, if we don't want anyone to use it anymore.

File: beetsplug/replaygain.py (hunk 1 above)
Comment (sampsyo) [reply beetsplug/replaygain.py]: Or perhaps better yet:

```py
results = [Gain(orig.gain + gain_adjustment, orig.peak) for orig in results]
```

**Comment by sampsyo** [comment]
Awesome; thank you! This is a heroic effort that required lots of special domain expertise, so I wanted to offer extra appreciation for getting this right. :sparkles:

**Pull request merged**